### PR TITLE
feat: select input token for pay with any token

### PR DESCRIPTION
**Pull request by JackShort**
* this allows you to select and updates the input token
* does not actually update price

### CODE REVIEW

File: src/nft/components/bag/tokenSelector/BagTokenSelectorModal.tsx [hunk 1]
```diff
@@ -90,7 +100,16 @@ export const BagTokenSelectorModal = ({ overlayClick }: { overlayClick: () => vo
         </TitleRow>
         <TokenSelectorContainer>
           {currencies.map((currency) => {
-            return <CurrencyRow key={currency.isToken ? currency.wrapped.address : currency.name} currency={currency} />
+            return (
+              <CurrencyRow
+                key={currency.isToken ? currency.wrapped.address : currency.name}
+                currency={currency}
+                selected={
+                  (!selectedCurrency && currency.isNative) || (!!selectedCurrency && selectedCurrency.equals(currency))
```
Comment (cbachmeier) [inline src/nft/components/bag/tokenSelector/BagTokenSelectorModal.tsx:108]: is the check for `!!selectedCurrency` needed?

File: src/nft/components/bag/tokenSelector/CurrencyRow.tsx [hunk 2]
```diff
@@ -4,11 +4,13 @@ import Column from 'components/Column'
...
 import Row from 'components/Row'
 import useCurrencyBalance from 'lib/hooks/useCurrencyBalance'
+import { Check } from 'react-feather'
 import styled from 'styled-components/macro'
 import { ThemedText } from 'theme'
 
 const TokenRow = styled(Row)`
   padding: 8px 0px;
+  gap: 12px;
   justify-content: space-between;
```
Comment (cbachmeier) [inline src/nft/components/bag/tokenSelector/CurrencyRow.tsx:14]: should probably have `cursor: pointer;` to show the row is interactible

**Review by cbachmeier — APPROVED** [review]
LGTM minor comments

File: src/nft/components/bag/tokenSelector/BagTokenSelectorModal.tsx (hunk 1 above)
Comment (JackShort) [reply src/nft/components/bag/tokenSelector/BagTokenSelectorModal.tsx:108]: surprisingly yes

**Pull request merged**